### PR TITLE
Emit typed event when preview is requested

### PR DESCRIPTION
**Pull request by juliusknorr**
Adds a typed event in order to replace the deprecated event that is already in place.

### CODE REVIEW

File: tests/lib/Preview/GeneratorTest.php [hunk 1]
```diff
@@ -109,7 +115,7 @@ public function testGetCachedPreview() {
 			->with($this->equalTo('256-256.png'))
 			->willReturn($previewFile);
 
-		$this->eventDispatcher->expects($this->once())
+		$this->legacyEventDispatcher->expects($this->once())
```
[the remark this answers is not in the document]
Comment (juliushaertl) [reply tests/lib/Preview/GeneratorTest.php:119]: Added

**Review by nickvergessen — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:

File: lib/public/Preview/BeforeFetchPreviewEvent.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+namespace OCP\Preview;
+
+use OCP\Files\Node;
+
+/**
+ * @since 25.0.1
+ */
+class BeforeFetchPreviewEvent extends \OCP\EventDispatcher\Event {
```
Comment (ChristophWurst) [inline lib/public/Preview/BeforeFetchPreviewEvent.php:34]:
```suggestion
class BeforePreviewFetchedEvent extends \OCP\EventDispatcher\Event {
```

**Review by ChristophWurst — APPROVED** [review]
:+1: :thinking:

File: lib/public/Preview/BeforeFetchPreviewEvent.php (hunk 2 above)
Comment (juliushaertl) [reply lib/public/Preview/BeforeFetchPreviewEvent.php]: Renamed :)

**Review by ChristophWurst — APPROVED** [review]

**Comment by ChristophWurst** [comment]
This event is missing on https://docs.nextcloud.com/server/latest/developer_manual/basics/events.html#available-events

**Pull request merged**